plugins/tileacq.py

```diff
@@ -285,13 +428,144 @@ def _get_fov(self, sd):
                 # compensate for binning
                 binning = ccd.binning.value
                 pxs = [p / b for p, b in zip(pxs, binning)]
-
                 return (shape[0] * pxs[0], shape[1] * pxs[1])
+
+            elif isinstance(sd, RepetitionStream):
+                # CL, Spectrum, AR
+                ebeam = sd.emitter
+                global_fov = (ebeam.shape[0] * ebeam.pixelSize.value[0],
+                              ebeam.shape[1] * ebeam.pixelSize.value[1])
+                l, t, r, b = sd.roi.value
+                fov = abs(r - l) * global_fov[0], abs(b - t) * global_fov[1]
+                return fov
             else:
-                raise TypeError("Unsupported Stream")
+                raise TypeError("Unsupported Stream ", sd)
         else:
             raise TypeError("Unsupported object")
 
+    def _cancel_acquisition(self, future):
+        """
+        Canceler of acquisition task.
+        """
+        logging.debug("Canceling acquisition...")
+
+        with future._task_lock:
+            if future._task_state == FINISHED:
+                return False
+            future._task_state = CANCELLED
+            future.running_subf.cancel()
+            logging.debug("Acquisition cancelled.")
+
+        return True
+
+    STITCH_SPEED = 1e-8  # s/px
+    MOVE_SPEED = 1e3  # s/m
+
+    def estimate_time(self, remaining=None):
```

Comment:
I'd recommend it returns the estimated duration, and add time.time() to every call. It's clearer than returning the time in absolute value.